[SDK-3025] Swift Docs - In-App Messages (#4855)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/swift/in-app_messaging/customization/handling_in_app_display.md

## Hiding the status bar during display

For `Full`, `FullImage` and `HTML` in-app messages, the SDK will hide the status bar by default. For other types of in-app messages, the status bar is left untouched. Using the `PresentationContext` object in your `inAppMessage(_:prepareWith:)` delegate method, you can configure the behavior of the status bar by setting the `statusBarHideBehavior` property. This field takes one of the following values:

| Status Bar Hide Behavior            | Description                                                                           |
| ----------------------------------- | ------------------------------------------------------------------------------------- |
| `.auto`                             | The message view decides the status bar hidden state.                                 |
| `.hidden`                           | Always hide the status bar.                                                           |

> | `.visible`                          | Always display the status bar.                                                        |

Reviewer: ```suggestion
| `.visible`                          | Always display the status bar.                                                        |
{: .reset-td-br-1 .reset-td-br-2}
```

@josh-mccrowell-braze, do we need to add those special directives each time we display a table? I've seen them used on other tables.

Reviewer 2: @lowip : Yes, probably a good idea. They keep line wrapping from breaking words.

## Logging impressions and clicks

> Logging in-app message impressions and clicks is not automatic when you implement completely custom handling (i.e., you circumvent Braze's in-app message display by returning `.discard` in your `inAppMessage(_:displayChoiceForMessage:)`). If you choose to implement your own UI using our in-app message models, you must log analytics with the following methods on the `ABKInAppMessage` class:

Reviewer 3: ```suggestion
Logging in-app message impressions and clicks is not automatic when you implement completely custom handling (i.e., you circumvent Braze's in-app message display by returning `.discard` in your `inAppMessage(_:displayChoiceForMessage:)`). If you choose to implement your own UI using our in-app message models, you must log analytics with the following methods on the `Braze.InAppMessage` class:
```

This block of text is outside any Swift vs ObjC tabs, so some of the syntax (e.g. `.discard`) are only in Swift. I think this is OK since it saves from diverging everywhere

| `.now`                              | The message will be displayed immediately.                                            |
| `.later`                            | The message will be not be displayed and will be placed back on the top of the stack. |
| `.discard`                          | The message will be discarded and will not be displayed.                              |
{: .reset-td-br-1 .reset-td-br-2}

You can use the `inAppMessage(_:prepareWith:)` delegate method to add in-app message display logic, customize in-app messages before Braze displays them, or opt-out of Braze's in-app message display logic and UI entirely.

## Overriding in-app messages before display

If you would like to alter the display behavior of in-app messages, you should add any necessary display logic to your `inAppMessage(_:prepareWith:)` [delegate method](https://braze-inc.github.io/braze-swift-sdk/documentation/brazeui/brazeinappmessageuidelegate/inappmessage(_:preparewith:)-11fog). For example, you might want to display the in-app message from the top of the screen if the keyboard is currently being displayed, or take the in-app message data model and display the in-app message yourself.

> If the in-app message campaign is not displaying when the session has been started, make sure you have the necessary display logic added to your `inAppMessage(_:prepareWith:)` delegate method. This allows the in-app message campaign to display from the top of the screen even if the keyboard is being displayed.

Reviewer: ```suggestion
```

Not relevant anymore.

## Hiding the status bar during display

> For `Full` and `HTML` in-app messages, the SDK will attempt to place the message over the status bar by default. However, in some cases, the status bar may still appear on top of the in-app message. Using the `PresentationContext` object in your `inAppMessage(_:prepareWith:)` delegate method, you can configure the behavior of the status bar by setting the `statusBarHideBehavior` property. This field takes one of the following values:

Reviewer: ```suggestion
For `Full`, `FullImage` and `HTML` in-app messages, the SDK will hide the status bar by default. For other types of in-app messages, the status bar is left untouched. Using the `PresentationContext` object in your `inAppMessage(_:prepareWith:)` delegate method, you can configure the behavior of the status bar by setting the `statusBarHideBehavior` property. This field takes one of the following values:
```
